mlflow/deployments/databricks/__init__.py

```diff
@@ -1,39 +1,138 @@
+        """
         raise NotImplementedError
 
     def list_deployments(self, endpoint=None):
+        """
+        .. warning::
+
+            This method is not implemented for `DatabricksDeploymentClient`.
+        """
         raise NotImplementedError
 
     def get_deployment(self, name, endpoint=None):
+        """
+        .. warning::
+
+            This method is not implemented for `DatabricksDeploymentClient`.
+        """
         raise NotImplementedError
 
+    def _call_endpoint(
+        self,
+        *,
+        method: str,
+        prefix: str = "/api/2.0",
+        route: Optional[str] = None,
+        json_body: Optional[Dict[str, Any]] = None,
+        timeout: int = MLFLOW_HTTP_REQUEST_TIMEOUT.get(),
+    ):
+        call_kwargs = {}
+        if method.lower() == "get":
+            call_kwargs["params"] = json_body
+        else:
+            call_kwargs["json"] = json_body
+
+        response = http_request(
+            host_creds=get_databricks_host_creds(self.target_uri),
+            endpoint=posixpath.join(prefix, "serving-endpoints", route or ""),
+            method=method,
+            timeout=timeout,
+            raise_on_status=False,
+            retry_codes=MLFLOW_DEPLOYMENT_CLIENT_REQUEST_RETRY_CODES,
+            **call_kwargs,
+        )
+        augmented_raise_for_status(response)
+        return AttrDict(response.json())
+
     def predict(self, deployment_name=None, inputs=None, endpoint=None):
-        raise NotImplementedError("TODO")
+        """
+        TODO
+        """
+        return self._call_endpoint(
+            method="POST",
+            prefix="/",
+            route=posixpath.join(endpoint, "invocations"),
+            json_body=inputs,
+            timeout=MLFLOW_DEPLOYMENT_PREDICT_TIMEOUT.get(),
+        )
 
     def create_endpoint(self, name, config=None):
-        raise NotImplementedError("TODO")
+        """
+        TODO
+        """
+        config = config.copy() if config else {}  # avoid mutating config
+        extras = {}
+        for key in ("tags", "rate_limits"):
+            if tags := config.pop(key, None):
+                extras[key] = tags
+        payload = {"name": name, "config": config, **extras}
+        return self._call_endpoint(method="POST", json_body=payload)
 
     def update_endpoint(self, endpoint, config=None):
-        raise NotImplementedError("TODO")
+        """
+        TODO
+        """
+        return self._call_endpoint(
+            method="PUT", route=posixpath.join(endpoint, "config"), json_body=config
+        )
 
     def delete_endpoint(self, endpoint):
-        raise NotImplementedError("TODO")
+        """
+        TODO
+        """
+        return self._call_endpoint(method="DELETE", route=endpoint)
 
     def list_endpoints(self):
-        raise NotImplementedError("TODO")
+        """
+        TODO
+        """
+        return self._call_endpoint(method="GET")
```

Comment:
Can we extract the `endpoints` field out of the response and return just that part?

Reply:
Great idea!